### ISSUE TITLE
@enum not working with pg-many-to-many plugin

### ISSUE THREAD

**Issue by hos** [graphile/graphile-engine#640]
When using `@enum` tags with `@graphile-contrib/pg-many-to-many`, I get 
`pgField call omits options.pgType for a leaf type; certain tweaks may not be applied!`.

SQL:
```sql
drop schema app_public cascade;
create schema app_public;
create table app_public.request_statuses (status text primary key);
insert into request_statuses values ('sent'), ('approved'), ('declined');
comment on table request_statuses is E'@enum';

create table members(id int generated by default as identity primary key);

create table app_public.cashout_requests(
  id int generated by default as identity primary key,
  member_id int not null references members,
  amount decimal(12, 2) not null,
  status text not null references request_statuses default 'sent',
  updated_at timestamptz not null default now(),
  created_at timestamptz not null default now()
);
```

Command:
```sh
mkdir enum-issue && cd enum-issue
yarn init -y
yarn add postgraphile@4.8.0 @graphile-contrib/pg-many-to-many
node ./node_modules/.bin/postgraphile -c test -s app_public --append-plugins @graphile-contrib/pg-many-to-many
```

**Comment by mattbretl** [comment]
This was caused by a bug in the `@graphile-contrib/pg-many-to-many` plugin.

Fixed in https://github.com/graphile-contrib/pg-many-to-many/pull/49 and included in version 1.0.0. :tada:

**Comment by benjie** [comment]
Oh awesome thanks Matt! I was planning to look into this on Friday; now I can release 4.8.0 🙌
